sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/UserAgentContainer.java

```diff
@@ -8,7 +8,7 @@
  */
 public class UserAgentContainer {
 
-    private static final int MAX_SUFFIX_LENGTH = 64;
+    private static final int MAX_SUFFIX_LENGTH = 192;
```

Comment:
I don't remember if there was a reason we choose 64. it is worth checking if user agent header on the service side, and kusto tables have any limit.

Reply:
64 is also used in .Net - I assume that is the reason. Kusto has a 1MB limit on Strings. Keeping it limited makes sense - but 64 characters is too restrictive when trying to squeeze in SparkConnector info + custom applicationName suffix.

I have tested the 192 characters end-to-end and it works.

Reply:
64 is used mostly for the user application identifier, not counting the SDK information, in this case I guess that the increase should be valid or maybe the max would be 64 + the base suffix (spark version)

Reply:
Yes - along those lines. The Spark prefix contains the executor name - which is variable - so I chose a sufficiently large value that should allow around 64 characters for application while still keeping the total string length below 255 chars. But given that it is the suffix I don't think it matters functionally - only reason for restriction is to not allow users to spam us with too much data in Kusto.

Reply:
Actually there is a hard constraint - Direct transport will throw an error if I force the userAgent to be larger than 255 characters. So I guess we have to set a hard limit on 255 chars. Will push another iteration.